### DISCUSSION TITLE
No alert fires even when MySQL replication lag has been high for a long time

### DISCUSSION THREAD

**Discussion by czxin788** [ccfos/nightingale#1411] [Q&A]
1、主从数据库延迟已经有21天了，夜莺没报警
![image](https://user-images.githubusercontent.com/13861904/222650675-dddc28d0-f0ee-4613-a4aa-f9fce62e075b.png)
2、查看夜莺不报警的原因：
![image](https://user-images.githubusercontent.com/13861904/222650762-b35ab51c-9600-40ce-aa03-8f0bb6fb986b.png)

3、确实查不出结果来，原因是mysql_slave_status_sql_delay没有这个key，只有mysql_slave_status_seconds_behind_master 这个key

![image](https://user-images.githubusercontent.com/13861904/222650966-0963c761-2ca6-42d8-ac33-9c49442a7fc9.png)
![image](https://user-images.githubusercontent.com/13861904/222651009-357881f6-1609-450f-92cf-21d761dc7a65.png)

4、于是自己手工创建mysql主从同步延迟告警
![image](https://user-images.githubusercontent.com/13861904/222651171-eb7a177d-b8c4-4832-9304-1e87fe8ec57f.png)
5、终于有告警了
![image](https://user-images.githubusercontent.com/13861904/222651270-56228a59-2251-4081-b360-86282861fde1.png)

6、我用的是categraf采集的数据，不知道为啥官方的告警模板不好用，目前我知道的是还有磁盘利用率没有告警，只有磁盘趋势告警，预计4小时写满，这个告警不如磁盘利用率90%告警有用

**Comment by wushengwusheng** [comment]
disk_used_percent
这个指标你没有吗？同categraf

**Comment by czxin788** [comment]
disk_used_percent指标有，但是告警规则里面没有针对disk_used_percent做告警，只针对predict_linear(disk_free[1h], 4*3600) < 0做的告警如下：
![image](https://user-images.githubusercontent.com/13861904/223000808-f11b62ac-e0b9-4e5a-acb2-c59a3c05c23d.png)

这么做的问题是，我有个机器磁盘都满了，“硬盘-预计再有4小时写满”没有告警：

于是，我手工加了个disk_used_percent的触发器：
![image](https://user-images.githubusercontent.com/13861904/223001067-4543f889-0ca1-4b21-98d6-44b02ac166ee.png)

我的问题是，建议官方模板把这种通用的告警触发器加好。
